spec/lib/payment_calculator/ecf/contract/uplift_payment_calculations_spec.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe PaymentCalculator::ECF::Contract::UpliftPaymentCalculations do
+  let(:dummy_class) { Class.new { include PaymentCalculator::ECF::Contract::UpliftPaymentCalculations } }
+  let(:call_off_contract) { create(:call_off_contract) }
+  let(:calculator) { dummy_class.new({ contract: call_off_contract }) }
+
+  context "when condition" do
+    it "returns the uplift_payment_per_participant" do
+      expect(calculator.uplift_payment_per_participant).to eq(100)
+    end
+
+    it "returns the expected uplift_payment_per_participant for a each event type" do
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :started)).to eq(100)
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :retained_1)).to eq(0)
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :retained_2)).to eq(0)
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :retained_3)).to eq(0)
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :retained_4)).to eq(0)
+      expect(calculator.uplift_payment_per_participant_for_event(event_type: :completed)).to eq(0)
+    end
+
+    it "returns a capped uplift_payment_for_events when the total exceeds 5% of the total contract value" do
+      expect(calculator.uplift_payment_for_event(uplift_participants: 10_000, event_type: :started)).to eq(99_500)
+    end
```

Comment:
Ok, so we get 10k participants. The uplift is 99.5k. For that to be 5%, the total contract value needs to be 1990k. 

Can we make those numbers... rounder somehow? Like, set up the test contract so that total value is 1 million, and show that if that's the case, the max uplift is 50k? Or something like that?

As it stands, the test shows that the value matches the expectations of the test writer, but does little to document why it needs to be that.

I guess we could make it more explicit - calculate total contract value in this test, and show that we get 5% of it - rather than hard-coding the value. Thoughts?

Reply:
Why? This is testing something that takes a number and multiplies it by 0.05. I think enough time has been spend on this without building even more testing around it.

Reply:
Because when I read

> it "returns a capped uplift_payment_for_events when the total exceeds 5% of the total contract value"

I want to verify that the test body matches the description. And veryfying that `99_500` is 5% of the contract value is not easy.